### ISSUE TITLE
Hide button until user selects payment method?

### ISSUE THREAD

**Issue by NicolasCapparelli** [francoislevesque/vue-braintree#21]
It's kind of confusing for the user to have the Pay button there even before the user enters their payment method

**Comment by ultrono** [comment]
Yep agree, I just made a local component from `https://github.com/francoislevesque/vue-braintree/blob/master/src/Payment.vue` and imported that into my own project, for speed.

I then added a prop: 

```
btnEnabled: {
   type: Boolean,
   default: true
},
```

and then updated the template:

```
<button @click="submit" :class="btnClass" :disabled="!btnEnabled">
```

I then used the events the component fires anyway to enable and disable the button. 

I also used the existing `load` event emitted by the original component to toggle the disabled attribute on the button. The loaded event passes the dropin instance and can be used to listen to native dropin events (see https://developers.braintreepayments.com/guides/drop-in/customization/javascript/v3#events) i.e. 

```
onLoaded (instance) {
   instance.on('paymentMethodRequestable', function (event) {}
   instance.on('noPaymentMethodRequestable', function () {}
}
```

**Comment by steffendybvik** [comment]
Or, you could just override the slot with your own button.

`    <v-braintree
      @load="onLoad"
    >
      <template v-slot:button="slotProps">
        <v-btn
          v-if="loaded"
          class="mt-5"
          block
          color="success"
          @click="slotProps.submit"
          >Pay</v-btn
        >
        <div v-else></div>
      </template>
    </v-braintree>`
